packages/utilities/src/lib/utilityTypes.ts

```diff
@@ -72,3 +72,91 @@ export type Awaited<T> = PromiseLike<T> | T;
  * Type union for the full 2 billion dollar mistake in the JavaScript ecosystem
  */
 export type Nullish = null | undefined;
+
+/**
+ * Similar to the built in {@link NonNullable}, but properly removes `null` from all keys in the class or interface
+ * This does not recurse deeply, for that use {@link DeepRequired}
+ */
+export type NonNullableClass<T = unknown> = {
+	[P in keyof T]: NonNullable<T[P]>;
+};
+
+/**
+ * An object that is non nullable, to bypass TypeScript not easily working with {@link Record}<{@link PropertyKey}, unknown> in various instances.
+ */
+export interface NonNullObject {}
+
+/**
+ * Any object at all. This can be used if you need to assign the most generic type of object to a type.
+ */
+export type AnyObject = Record<PropertyKey, unknown> | NonNullObject;
+
+/**
+ * Gets all the keys (as a string union) from a type `T` that match value `V`
+ * @example
+ * interface Sample {
+ * 	id: string;
+ * 	name: string | null;
+ * 	middleName?: string;
+ * 	lastName: string;
+ * 	hobbies: readonly string[];
+ * }
+ *
+ * type BB = KeyOfType<Sample, string>;
+ * // Expected:
+ * // "id" | "lastName"
+ */
+export type KeyOfType<T, V> = {
```

Comment:
Rename to PropertiesOfType

Reply:
Then why did you name it like this in Skyra. Also in https://github.com/piotrwitek/utility-types it is named [`PickByValue`[(https://github.com/piotrwitek/utility-types#pickbyvaluet-valuetype) which is leagues better than `PropertiesOfType` because this doesn't get the properties of the type, this gets the properties of a type as based on another type.